### PR TITLE
Adding BC check

### PR DESCRIPTION
**Pull request by Nyholm**
This is a super strict check =)

### CODE REVIEW

**Comment by Nyholm** [comment]
Blocked by https://github.com/Nyholm/roave-bc-check-docker/pull/15

=)

**Comment by Nyholm** [comment]
Im done with this PR now. 

It will run [a BC check](https://github.com/Roave/BackwardCompatibilityCheck). I also added a file to ignore some BC warnings. 

Ie, updating the value of a public static property is technically a BC break. But that should be ignore here since all our "random data" is a public static property.

File: .github/workflows/bc-check.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+on:
+  pull_request:
+  push:
```
Comment (sanmai) [inline .github/workflows/bc-check.yml:4]: Should probably match `main` branch wdyt?

Comment (Nyholm) [reply .github/workflows/bc-check.yml:4]: Yes, Correct. 

I see that you took a different direction than the one I suggested in #26. 

I've updated the PR. Thank you

**Review by sanmai — APPROVED** [review]

**Comment by pimjansen** [comment]
Thanks. Will really help us keep the stability there!

**Comment by Nyholm** [comment]
Thank you for merging. 

Yes, indeed. This will be super helpful to see if we accidentally break BC.

**Pull request merged**